### PR TITLE
[POAE7-1216] convert filter push down info in json format

### PR DESCRIPTION
**Pull request by yma11**
Test plan - (Please fill in how you tested your changes)
N/A

General Changes
* Collect filter push down info and generate Json string for RelNode

### CODE REVIEW

**Comment by yma11** [comment]
@winningsix @jikunshang, please take a review.

File: presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdownDelegation.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    }
+
+    private static JsonNode getJsonFilterNode(FilterNode filterNode)
+    {
+        ObjectMapper objectMapper = new ObjectMapper();
+        ObjectNode objectNode = objectMapper.createObjectNode();
+        objectNode.put("id", filterNode.getId().toString());
+        objectNode.put("relOp", "LogicalFilter");
+        ObjectNode conditionNode = objectMapper.createObjectNode();
+        if (filterNode.getPredicate() instanceof CallExpression) {
```
Comment (jikunshang) [inline presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdownDelegation.java:81]: don't support multi filter expression?

**Review by jikunshang — COMMENTED** [review]
possible to add a ut? or attach some real output?

File: presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdownDelegation.java (hunk 1 above)
Comment (yma11) [reply presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdownDelegation.java:81]: Yes. Currently this is just for query like select c_customer_id from hive.tpcds1gorc.customer where c_current_addr_sk=7727 for our prototype. It's not a good idea to follow this convert method in future.

**Comment by winningsix** [comment]
A few high level points:
1. Key logic from optimizer is in ```ApplyConnectorOptimization``` which triggers the optimize method. In the end, cider will need an optimizer converting nodes. (TODO in other PR)
2. We need to broadcast the jsonString to each node. You can touch Yang's ongoing PR. (TODO in other PR)

Anyway, I am +1 on this PR for now.

**Pull request merged**